src/components/application_manager/src/commands/command_request_impl.cc

```diff
@@ -788,6 +790,78 @@ bool CommandRequestImpl::IsResultCodeUnsupported(
          (first.is_unsupported_resource && second.is_unsupported_resource);
 }
 
+void CommandRequestImpl::AddComponentInfoToMessage(
+    smart_objects::SmartObjectSPtr& response) const {
+  using NsSmartDeviceLink::NsSmartObjects::SmartObject;
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  if (!response) {
+    LOG4CXX_WARN(logger_, "Invalid message object");
+    return;
+  }
+
+  const uint32_t app_connection_key = connection_key();
+  const ApplicationSharedPtr app =
+      application_manager_.application(app_connection_key);
+  if (!app) {
+    LOG4CXX_WARN(logger_, "Invalid connection_key: " << app_connection_key);
+    return;
+  }
+
+  const SmartObject* app_type = app->app_types();
+  if (!app_type) {
+    LOG4CXX_WARN(logger_, "Empty application types array!");
+    return;
+  }
+
+  if (NsSmartDeviceLink::NsSmartObjects::SmartType_Array !=
+      app_type->getType()) {
+    LOG4CXX_WARN(logger_, "Application types are not an array!");
+    return;
+  }
+
+  const SmartObject& app_type_so = app_type->getElement(0);
+  if (NsSmartDeviceLink::NsSmartObjects::invalid_object_value == app_type_so) {
+    LOG4CXX_WARN(logger_, "Invalid object for component type!");
+    return;
+  }
+
+  const int64_t app_hmi_type = app_type_so.asInt();
+  const std::string& component_name = AppHMITypeToString(
+      static_cast<mobile_apis::AppHMIType::eType>(app_hmi_type));
+  const std::string component_info =
+      component_name + " component does not respond";
```

Comment:
@istoilovgithub please make description for function more clear. It is not clear from name and description this this function should be called only if some component does not respond.